File: src/main/java/io/github/trainb0y/simplehud/elements/PositionElement.java
package io.github.trainb0y.simplehud.elements;

import net.minecraft.client.MinecraftClient;
import net.minecraft.util.math.BlockPos;

import java.util.List;

public class PositionElement extends Element {
    public PositionElement() {
        super();
    }

    public PositionElement(int x, int y, boolean enabled) {
        super(x, y, enabled);
    }

    public Object[] getArgs(MinecraftClient client) {
        BlockPos pos = client.player != null ? client.player.getBlockPos() : new BlockPos(0, 0, 0);
        return List.of(pos.getX(), pos.getY(), pos.getZ()).toArray();
    }

    @Override
    public String getKey() {
        return "element.simplehud.position";
    }
}

File: src/main/kotlin/io/github/trainb0y/fabrihud/elements/PositionElement.kt
package io.github.trainb0y.fabrihud.elements

import net.minecraft.client.MinecraftClient
import net.minecraft.util.math.BlockPos

// HUD Element displaying position of player
class PositionElement : Element() {
	override fun getArgs(client: MinecraftClient): List<Any?> {
		val pos = client.player?.blockPos ?: BlockPos(0, 0, 0)
		return listOf(pos.x, pos.y, pos.z)
	}

	override val key = "element.fabrihud.position"
}